housekeeping/src/main/java/com/hotels/housekeeping/service/impl/FileSystemHousekeepingService.java

```diff
@@ -109,11 +109,14 @@ private void housekeepPath(LegacyReplicaPath cleanUpPath) {
   public void cleanUp(Instant referenceTime) {
     try {
       Pageable pageRequest = new PageRequest(0, fetchLegacyReplicaPathPageSize);
-      Page<LegacyReplicaPath> page;
-      do {
+      Page<LegacyReplicaPath> page = legacyReplicaPathRepository
```

Comment:
OK, so I find this confusing and the Spring docs and examples don't make it much clearer but from what I can understand is that we: try process a page, we then ask that page if there are other pages (`page.hasNext()`) and if so, we create a new pageRequest (`pageRequest.next()`) to get these results out into a page etc.

Reply:
yes